WordPress/gutenberg
# docgen: Omit docblocks with private tag

Closes #14294

This pull request seeks to enhance `@wordpress/docgen` to omit symbols which include a `@private` tag in their JSDoc. In turn, this was used to omit the reference to `wp.blocks.children` from the Blocks API documentation.

**Open Questions:**

There might be a question as to how or if we choose to omit deprecated APIs from the documented output, as in this case the API is technically deprecated. If we choose to omit this deprecated API, should we do the same for all (or should it be an option of `docgen` to omit deprecated APIs?). Otherwise, should this API not be marked as private and instead simply marked as deprecated, but allowed to be included in the documented output?

**Testing Instructions:**

Verify there are no changes from running `npm run docs:build`.

Ensure unit tests pass:

```
npm run test-unit
```
